docs/config-and-instances.md

```diff
@@ -0,0 +1,340 @@
+or computed values like bus addresses.
+
+## Creation
+To create a VM definition, the user needs to `POST` a `Config`.
+
+## Starting a VM, the `/start` sub-resource
+To create an instance from a `Config` a user has to `GET` the `start`
+sub-resource of a `Config` instance.
+
+## Stopping a VM, the `/stop` sub-resource
+To stop a VM, the user can `GET` the `stop` sub-resource of the associated
+`Config` object.
+
+**Note:** The semantics of a `DELETE` on the `VM` object of a `Config` still
+need to be defined.
+
+## Changing a `Config`
+Changes to a `Config` instance can be performed through the usual `PATH` calls.
+Chaging a `Config` does not lead to an automatic propagation of the change to
+the instance.
+However, in future there could be a specific metadata or action to trigger
+such a propagation automatically.
+
+
+### Example Flow
+
+Let's start with a simple example to get a better feeling for the realtionship
+between the objects.
+
+```
+$c1        $i1        $i2
+POST
+/start     POST
+/stop      DELETE
+PATCH
+/start                POST
+/stop                 DELETE
+```
+
+
+1. `POST` $c1 config
+2. `GET` $c1/start
+   → creates instance `POST` $i1
+3. `GET` $c1/stop
+   → `DELETE` $i1
+4. `PATCH` $c3
+5. `GET` $c1/start
+   → creates instance `POST` $i2
+6. `GET` $c1/stop
+   → deletes instance `DELETE` $i2
+
+In this example, a the user defines a VM, then starts it and an instance is
+getting created. Once the user stops it, the instance is getting removed.
+After changing the `Config` (in step 4), he starts and stops the VM again.
+
+This is pretty straight forward.
+
+
+### Example Flow with persistent VM
+
+In the previous example, the VM is always regenerated from the high-level
+`Config` object. This has the implication that the `VM` config can change
+between two runs. This might not be desireable in some scenarios.
+
+To solve this problem, the user can signal the system, to create a `Config`
+'snapshot' of a `VM` object, whenever it is getting created.
+The flag `instanceConfig` can be set on a `Config`, to tell the system
+to create a new `Config` object whenever a VM is started based on this
+`Config`.
+
+By default this `Config` is only created when the VM is launched. Thus changes
+performed on the `VM` object at runtime will not be reflected in the created
+`Config`. But the switch `autoSave` can be used to automatically propagate
+changes done to the `VM` object at runtime, to the `Config` object.
+
+But let's look at an example to grasp this:
+
+**Note:** The only relevant endpoints for the initial concept are `/start` and
+`/stop`. Additional endpoints (`/save` and `/snapshot`) are just to illustrate
+how the mechanism could eventually be used for other functionality as well.
+
+```
+$c1        $c2        $c3        $i1        $i2        $i3       $i4
+POST
+/start                           POST
+           POST
+/stop                            DELETE
```

Comment:
In the example I was starting c2 to highlight that c2 can now be used to re-create the very same vm as before.
Starting c1 again could lead to a different expansion (assuming that we do expansion at that level [from c --> i])